### ISSUE TITLE
False negative for unnecessary_map_or (but not with rust-analyzer)

### ISSUE THREAD

**Issue by BaumiCoder** [rust-lang/rust-clippy#15683]
### Summary

With clippy 0.1.89 (29483883ee 2025-08-04) executed with cargo 1.89.0 (c24e10642 2025-06-23), an instance of an `unnecessary_map_or` is not reported. When using rust-analyzer v0.3.2593 in VSCodium 1.104.06131 to run clippy on save, I get the instance of `unnecessary_map_or`.

### Lint Name

unnecessary_map_or

### Reproducer

I tried this code:

```rust
let path = PathBuf::from("./example.txt");
path.extension()
    .map_or(true, |ext| ext != "editorconfig" && ext != "license")
```

A full example with VSCodium / VSCode configuration is my project at this commit:
https://codeberg.org/BaumiCoder/ecformat/src/commit/345b116be7b2c7aee6f2d37057573ce75693cdef
There you can find the warning (inside VSCodium) in `src/editorconfig/charset/tests/test_utils.rs`.

I expected to see this happen:

Get a warning like this:

```
warning: this `map_or` can be simplified
  --> tests/../src/editorconfig/charset/tests/test_utils.rs:21:13
   |
21 | /             path.extension()
22 | |                 .map_or(true, |ext| ext != "editorconfig" && ext != "license")
   | |______________________________________________________________________________^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_map_or
   = note: `#[warn(clippy::unnecessary_map_or)]` on by default
help: use is_none_or instead
   |
22 -                 .map_or(true, |ext| ext != "editorconfig" && ext != "license")
22 +                 .is_none_or(|ext| ext != "editorconfig" && ext != "license")
   |
```

Instead, this happened:

I get no warning with `cargo clippy`, but at least VSCodium shows with rust-analyzer the warning.

### Version

```text
rustc 1.89.0 (29483883e 2025-08-04)
binary: rustc
commit-hash: 29483883eed69d5fb4db01964cdf2af4d86e9cb2
commit-date: 2025-08-04
host: x86_64-unknown-linux-gnu
release: 1.89.0
LLVM version: 20.1.7
```

**Comment by BaumiCoder** [comment]
The example is in test code. With `cargo clippy --all-targets`, I get the warning. If I understand the [CI example](https://github.com/rust-lang/rust-clippy?tab=readme-ov-file#travis-ci) correctly, that is the expected behavior. (Strange that rust-analyzer considers test code by default 🤔)
